geomesa-core/src/main/scala/org/locationtech/geomesa/core/index/Strategy.scala

```diff
@@ -163,6 +160,22 @@ trait Strategy {
       configureFeatureType(cfg, featureType)
 
       Some(cfg)
-    } else None
+    case mapAggregationQuery if mapAggregationQuery.getHints.containsKey(MAP_AGGREGATION_KEY) =>
```

Comment:
I feel like this needs to be refactored.  Doesn't have to be you.  

If you don't make this better, can you add a comment referencing this ticket: https://geomesa.atlassian.net/browse/GEOMESA-698 ?